src/Mechanic.Tests/Tests.fs

```diff
@@ -3,9 +3,31 @@ module Tests
 open FsCheck.Xunit
 open Swensen.Unquote
 open Mechanic
+open Mechanic.GraphAlg
 
 [<Property>]
 let ``Say says hello to whomever we greet`` greetee =
     let greeting = Say.hello greetee
     let expected = "Hello " + greetee
     greeting =! expected
+
+[<Property>]
+let ``Topological order alg`` (edges: list<int * int>) =
+    let rec haveCycleAcc edges acc =
+        let (edgesFrom, edgesRemain) = edges |> List.partition (fun (v,_) -> Set.contains v acc)
+        match edgesFrom with
+        | [] -> false
+        | _ ->
+        let nodes = edgesFrom |> Seq.map (fun (_,w) -> w) |> set
+        match Set.intersect nodes acc |> Set.isEmpty with
+        | true -> haveCycleAcc edgesRemain (acc + nodes)
+        | false -> true
+    let haveCycle (nodes: list<int>) edges = nodes |> Seq.exists (fun v -> haveCycleAcc edges (set [v]))
```

Comment:
If the set being created here is only one item, can you change the comparison in line 17 to just `(=)` rather than a Set operation (performance, readability)?

Reply:
Set is singleton only in first call of recursive function `haveCycleAcc`.